### PR TITLE
Fix issue that data is incomplete in full join of table model and add new ExchangeNode for table model

### PR DESCRIPTION
**Pull request by Beyyes**
## Description

As the title. 

Besides, 
1. This pr also fix the problem of explain when there the query has no devices of data partitions.
2. Add new ExchangeNode for table model
3. Add maxReservedMemory metric for join operator

<hr>

This PR has:
- [x] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [x] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/distribute/TableDistributedPlanGenerator.java [hunk 1]
```diff
@@ -495,6 +494,11 @@ public List<PlanNode> visitTableScan(TableScanNode node, PlanContext context) {
       }
     }
 
+    if (tableScanNodeMap.isEmpty()) {
+      node.setRegionReplicaSet(new TRegionReplicaSet());
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/distribute/TableDistributedPlanGenerator.java:498]: use the NOT_ASSIGNED singleton

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java [hunk 2]
```diff
@@ -307,17 +309,27 @@ private void prepareForNextDevice() {
     }
   }
 
-  private AlignedSeriesScanUtil constructAlignedSeriesScanUtil(DeviceEntry deviceEntry) {
+  private void constructAlignedSeriesScanUtil() {
+    DeviceEntry deviceEntry;
+
+    if (this.deviceEntries.isEmpty() || this.deviceEntries.get(this.currentDeviceIndex) == null) {
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java:315]: throw Exception for this.deviceEntries.get(this.currentDeviceIndex) == null

Comment (JackieTien97) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java:315]: No need to construct `AlignedFullPath` and `seriesScanUtil` for `this.deviceEntries.isEmpty()`

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/distribute/TableDistributedPlanGenerator.java (hunk 1 above)
Comment (Beyyes) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/distribute/TableDistributedPlanGenerator.java]: fix

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java (hunk 2 above)
Comment (Beyyes) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java]: fix

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**